### DISCUSSION TITLE
Can i use proxy only on extract

### DISCUSSION THREAD

**Discussion by lcexc** [streamlink/streamlink#4227] [Q&A] [answered]
(sorry to my poor english)
I always use streamlink to see live on afreecatv.com. Because of the GFW, I need use proxy.
Recently, I fonud the live stream can be access directly(cdn link?),  so only the extract need proxy, how can i use proxy only on extract ?
Is someone can help me? very thanks!

**Comment by bastimeyer** [comment, marked as answer]
If the resolved stream URL doesn't rely on any IP-specific authentication parameters, custom HTTP headers or cookies, or other non-standard streaming protocol workarounds that Streamlink needs to implement in order to get the stream output working, then you can simply let Streamlink output the resolved stream URL via [`--stream-url`](https://streamlink.github.io/cli.html#cmdoption-stream-url) (with a proxy config applied) instead of opening the stream and piping the data into the player (the default behavior). Then you can open the stream URL in your player without any proxy configs or even use it as input URL for Streamlink again.

Example:
```bash
STREAM_URL=$(streamlink --stream-url --http-proxy "$PROXY" "$INPUT" "$QUALITY")
"$PLAYER" "$STREAM_URL"
```

The [`--player-passthrough hls`](https://streamlink.github.io/cli.html#cmdoption-player-passthrough) parameter does the exact same thing as above without having to do any shell scripting or manual URL copying:
```bash
streamlink --player-passthrough hls --player "$PLAYER" --http-proxy "$PROXY" "$INPUT" "$QUALITY"
```

If any HTTP headers or cookies are however required for accessing the stream URL, then you have to figure this out via the player config, not Streamlink. In case of the Afreeca plugin, it uses a custom HLS implementation which filters out "preloading" segments, but that shouldn't cause any issues in this case here.
https://github.com/streamlink/streamlink/blob/3.0.3/src/streamlink/plugins/afreeca.py#L12-L22
